apps/management-app/src/components/Authenticated/CDO/ReportTypesDetail/index.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { yupResolver } from '@hookform/resolvers/yup';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate, useParams } from 'react-router-dom';
+import * as yup from 'yup';
+import ControlledTextField from '@/components/Common/ControlledTextField';
+import {
+  useDeleteReportTypesMutation,
+  useLazyGetReportTypeByIdQuery,
+  useUpdateReportTypeMutation,
+} from '@/store/api/generalManagementApiSlice';
+import { isApiErrorResponse } from '@/store/api/helper';
+import { showError, showSuccess } from '@/utils/toast';
+import FormInputSkeleton from '../FormInputSkeleton';
+import StaticActionBar from '../StaticActionBar';
+
+interface FormData {
+  name: string;
+}
+
+const ReportTypesDetail = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  const [getReportType, { isError }] = useLazyGetReportTypeByIdQuery();
+
+  useEffect(() => {
+    if (isError) navigate(-1);
+  }, [isError, navigate]);
+
+  const schema = useMemo(
+    () =>
+      yup.object({
+        name: yup.string().required("Report type's name is required"),
+      }),
+    [],
+  );
```

Comment:
should define this validation outside of component, on top for example

Reply:
The schema has component scope and should be wrapped inside its component.

Reply:
Updated